src/Microsoft.Extensions.Logging.Console/ConsoleLoggerProvider.cs

```diff
@@ -2,23 +2,109 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
 
 namespace Microsoft.Extensions.Logging.Console
 {
     public class ConsoleLoggerProvider : ILoggerProvider
     {
+        private readonly ConcurrentDictionary<string, ConsoleLogger> _loggers = new ConcurrentDictionary<string, ConsoleLogger>();
+
         private readonly Func<string, LogLevel, bool> _filter;
-        private readonly bool _includeScopes;
+        private readonly IConsoleLoggerSettings _settings;
 
         public ConsoleLoggerProvider(Func<string, LogLevel, bool> filter, bool includeScopes)
         {
+            if (filter == null)
+            {
+                throw new ArgumentNullException(nameof(filter));
+            }
+
             _filter = filter;
-            _includeScopes = includeScopes;
+            _settings = new ConsoleLoggerSettings()
+            {
+                IncludeScopes = includeScopes,
+            };
+        }
+
+        public ConsoleLoggerProvider(IConsoleLoggerSettings settings)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            _settings = settings;
+
+            if (_settings.ChangeToken != null)
+            {
+                _settings.ChangeToken.RegisterChangeCallback(OnConfigurationReload, null);
+            }
+        }
+
+        private void OnConfigurationReload(object state)
+        {
+            _settings.Reload();
+
+            foreach (var logger in _loggers.Values)
+            {
+                logger.Filter = GetFilter(logger.Name, _settings);
+                logger.IncludeScopes = _settings.IncludeScopes;
+            }
+
+            // The token will change each time it reloads, so we need to register again.
+            if (_settings?.ChangeToken != null)
```

Comment:
We might discuss this more: https://github.com/aspnet/specs/blob/master/notes/reloadable_options.md#open-issues

But this PR isn't going to reinvent the pattern for using `IChangeToken`. If we find a way to simplify the pattern, we'll do that independently and clean this up later.